sig-and-wg/sig-core/proposals/objects-store-proposal.md

```diff
@@ -0,0 +1,155 @@
+# Objects Store
+
+Created on 2019-01-03 by Lukasz Gornicki (@derberg).
+
+## Status
+
+Proposed on 2019-01-04
+
+## Motivation
+
+1. Have a generic solution for storing in s3 like storage any kind of object: zip file, markdown file, PNG or JS (JavaScript)
+2. Have a storage solution not locked to one vendor, like AWS S3 only, or like to Minio on-premise only
+3. Have a storage solution that not only stores the file but also exposes it directly to internet
+
+## Use case
+
+1. Storage for documentation and related images
+2. Storege for API specifications
+3. Storage for static client-side applications
+
+## Solution
+  
+![](assets/storage.svg)
+
+1. Location of the object is specified in the Object custom resource.
+2. ObjectStore controller fetches the object basing on the information given in the custom resource.
+3. The controller performs:
+    - Mutation of the object by communicating with mutation webhook specified in the custom resource
+    - Validation of the object by communicating with mutation webhook specified in the custom resource
+    - New file creation, if such file was referenced in the resource definition as a ConfigMap
+   If any of above operations failed, controller updates the resource with `ready: False` status  
```

Comment:
This line doesn't render well and shows up as a continuation of the previous bullet point.

Reply:
fixed